libraries/ms-common/inc/interrupt.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+// Shared interface elements across all devices for interrupts. Each device will also have internal
+// device specific functionality which exist in their respective interrupt_mcu.h files (do not
+// include that file directly instead reference it through a module such as gpio_it.h).
+
+#include <stdbool.h>
+
+// A generic interrupt callback with context.
+typedef void (*interrupt_callback)(void *context);
```

Comment:
I'm not a fan of passing in the calling function. That requires doing string comparisons and keeping track of what the function name is.

I'd rather have callbacks per source that can provide module-specific identifiers. For example, the timer callback might include the ID of the timer that expired and some context information. The GPIO callback might include the port/pin combination. The goal is ultimately for callbacks to run as quickly as possible.